File: problems/contains-duplicate-ii/sliding-window.cs
public class Solution {
    public bool ContainsNearbyDuplicate(int[] nums, int k) {
        for (var i = 0; i < nums.Length; i++)
        {
            var num = nums[i];
            var j = i + 1;
            while (j - i <= k && j < nums.Length)
            {
                if (j >= nums.Length) break;
                var num2 = nums[j];
                if (num == num2) return true;
                j++;
            }
        }

        return false;
    }
}

File: problems/max-consecutive-ones/sliding-window.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        window = []
        maxConsecutiveOnes = 0
        for num in nums:
            if num != 1:
                window = []
            else:
                window.append(num)
            
            maxConsecutiveOnes = max(maxConsecutiveOnes, len(window))
        return maxConsecutiveOnes